### ISSUE TITLE
Modal gets focused outline by mouse usage after reopen it (visual bug)

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#1971]
**BLOCKED**: First we need to refactor Modal + Flyout to use `<dialog>`. And before that, we need to finalize Flyout Navigation behaviour regarding `<dialog>`'s focus behaviour. Ref: #2928 

Description  
---
Modal gets visible outline although the user interaction occurred without keyboard navigation.

Notes
---
- https://porschedev.slack.com/archives/CT7AVHTTQ/p1657876679681649
- Maybe we should use `:focus-visible` only?
- Modal should never get an outline. It should be forwared to dismiss button directly?
- First we should use focus style for all components

To Reproduce
---
1. Go to https://designsystem.porsche.com/latest/components/modal/examples
1. Click on "Open Modal" button
1. Close Modal with "X" or ESC and open again with click on button
1. See error (Modal has visible focus outline)
1. Close Modal with "X" or ESC
1. See error ("Open Modal" button has visible focus outline)

Expected behavior
---
Modal and "Open Modal" button should only receive focused outline on keyboard usage.

Test Environment
---
Platform/Browser/Version: [MacOS Chrome 103.0.5060.114]

**Comment by AplusKminus** [comment]
We also noticed this focus border on the first click in some circumstances -- no need to reopen it.

Also: The steps to reproduce given in this issue do not reproduce the issue for me (Firefox on Mac). In this specific instance I have to press [TAB] and [SHIFT] + [TAB] after closing the modal and then re-open it.
